backend/core/llm_chain.py

```
@@ -3,490 +3,167 @@
 import hashlib
 import json
 import logging
-from typing import List, Optional, Dict, Any, Tuple
+import asyncio
+from typing import List, Optional, Dict, Any, Tuple, AsyncIterator
```

Ruff (0.12.2)

`typing.Tuple` imported but unused

Remove unused import: `typing.Tuple`

(F401)